### ISSUE TITLE
font-size-dependent displacement with specific font

### ISSUE THREAD

**Issue by wareya** [harfbuzz/harfbuzz#1595]
tl;dr: I'm rendering vertical Japanese text with NotoSansCJKjp-Regular.otf (https://github.com/googlei18n/noto-cjk/blob/master/NotoSansCJKjp-Regular.otf) and the placements of certain (but not all) punctuation symbols, especially opening "brackets" and "quotes" like 「, （, and 〝, are wrong. The wrongness is font size dependent. Advancement distance is correct. I'm using harfbuzz-2.3.1 and freetype-2.9.1.

24px:
![temp](https://user-images.githubusercontent.com/585488/53302316-134be180-382b-11e9-8cbe-a40e0bf645b7.png)

48px:
![temp](https://user-images.githubusercontent.com/585488/53302304-fa433080-382a-11e9-9eb7-74be962cbf3e.png)

The top prime is supposed to be pretty much exactly as close to the 田 as the bottom prime is.

The tiny red squares are the location of the cursor before/after drawing each glyph.

The exact values of ```y_offset/64.0``` are:

24px:
-15.390625
-20.171875
-20.828125

48px:
-20.093750
-38.578125
-41.140625

x_offset is -12 for 24px and -24 for 48px on every character in the string, as it should be.

I set the font size with ```FT_Set_Pixel_Sizes(fontface, 0, FONTSIZE)``` before calling ```hb_ft_font_create(fontface, NULL)```. This should (and does, according to examples, actual advancement distance, and ```x_offset```) cause ```y_offset``` and ```y_advance``` etc. to be scaled to pixels times 64 (i.e. 6 bits of fixed point subpixel accuracy).

```FT_LOAD_RENDER``` and ```FT_LOAD_VERTICAL_LAYOUT``` are being passed to ```FT_Load_Glyph```, and ```hb_buffer_set_direction``` is being passed ```HB_DIRECTION_TTB```. I'm not passing any features to hb_shape, but passing vkrn (for example) doesn't fix anything.

I'm not doing anything weird like scaling anything myself. "Basic" glyph position relative to the drawing cursor is being taken from freetype as. ```glyph_y = fontface->glyph->bitmap_top```. This is being added to ```y_offset/64.0``` to get the complete position relative to the cursor, as it should be.

The exact values of ```bitmap_top``` (this is from freetype) are:

24px:
4
19
21

48px:
9
37
41

The text I'm testing is "〝田〟". I picked it because, unlike other examples of broken punctuation, firefox doesn't have a hardcoded vertical representation form lookup for 〝 or 〟, and it still renders properly in firefox given the following css:

```
body {
    writing-mode: vertical-rl;
    font-family: "Noto Sans CJK JP", sans-serif;
    font-size: 48px;
}
```

firefox:
![image](https://user-images.githubusercontent.com/585488/53302379-98cf9180-382b-11e9-9928-b2126c97db9f.png)

Firefox uses harfbuzz for shaping, but seems to operate entirely in 1024\*64 units, not pixels\*64 units.

https://hg.mozilla.org/mozilla-central/file/1cacb0b1fe7b/gfx/thebes/gfxHarfBuzzShaper.cpp#l31

https://hg.mozilla.org/mozilla-central/file/1cacb0b1fe7b/gfx/thebes/gfxHarfBuzzShaper.cpp#l1581 (through line 1597)

**Comment by khaledhosny** [comment]
Looks like a FreeType issue/limitation, `hb-shape` output looks identical:
```
$ hb-shape NotoSansCJK-Regular.ttc 〝田〟 --direction=t --font-funcs=ft
[gid65143=0@-500,-880+0,-1000|gid27065=1@-500,-880+0,-1000|gid65144=2@-500,-880+0,-1000]
$ hb-shape NotoSansCJK-Regular.ttc 〝田〟 --direction=t --font-funcs=ot
[gid65143=0@-500,-880+0,-1000|gid27065=1@-500,-880+0,-1000|gid65144=2@-500,-880+0,-1000]
```

But the rendered `hb-view` output is different:
```
$ hb-view NotoSansCJK-Regular.ttc 〝田〟 --direction=t --font-funcs=ft
```
![ft](https://user-images.githubusercontent.com/93914/53302939-1bc50c00-386d-11e9-9071-0a0b43b3eb64.png)

```
$ hb-view NotoSansCJK-Regular.ttc 〝田〟 --direction=t --font-funcs=ot
```
![ot](https://user-images.githubusercontent.com/93914/53302942-22ec1a00-386d-11e9-8dc7-e621272f3bf7.png)

**Comment by behdad** [comment]
> `FT_LOAD_VERTICAL_LAYOUT` are being passed to `FT_Load_Glyph`

Try *not* doing that.

**Comment by wareya** [comment]
Still broken when using hb-ft without passing FT_LOAD_VERTICAL_LAYOUT to FT_Load_Glyph, only FT_LOAD_RENDER alone.

**Comment by behdad** [comment]
I've made fixes in vertical placement, and now ot and ft funcs return the same placement.
